[MERGED] Fix WorldEditAdditions help crash by ensuring command definitions have name field and preventing nil entries

Pivert opened the pull request:
**Problem Identified**: The crash was caused by the WorldEditAdditions mod registering commands in the `worldedit.registered_commands` table without setting the required `name` field in the command definition. When WorldEdit's help system tried to format help text, it attempted to concatenate `def.name` which was `nil`.

**Solution Implemented**: Modified the `register_command` function in `/workdir/luanti/mods/worldeditadditions/worldeditadditions_core/core/register_command.lua` to set `options.name = cmdname` before storing the command definition in the WorldEdit registered commands table. This ensures compatibility with WorldEdit's help system which expects the `name` field to exist.

The fix ensures that all WorldEditAdditions commands will now have the required `name` field, preventing the "attempt to concatenate field 'name' (a nil value)" crash when users use the WorldEdit help system.

Commit:
Fix WorldEditAdditions help crash by ensuring command definitions have name field and preventing nil entries

sbrl commented on worldeditadditions_core/core/register_alias.lua:
```diff
@@ -8,6 +8,11 @@ local function register_alias(cmdname_target, cmdname_source, override)
+		wea_c.registered_commands[cmdname_target] = nil
+		if minetest.global_exists("worldedit") then
+			worldedit.registered_commands[cmdname_target] = nil
+		end
```
did you find this was essential? aliases are only registered further down, so in theory this shouldn't be required..... but you're the one who put this PR together not me and WEA has a big codebase \:P

Pivert replied:
You're absolutely right - this cleanup is unnecessary here since we're in the error path before any registration has occurred. The cleanup code should be removed as it doesn't serve any purpose at this point in the function.

sbrl commented on worldeditadditions_core/core/register_alias.lua:
```diff
@@ -29,10 +34,18 @@ local function register_alias(cmdname_target, cmdname_source, override)
+			if source_def then
+				worldedit.registered_commands[cmdname_target] = source_def
+			end
+		end
```
a cleaner alternative to this would be to use `wea_c.fetch_command_def(cmdname_source)` and do a `nil` check, which references the WE command table automatically

sbrl commented on worldeditadditions_core/init.lua:
```diff
@@ -65,6 +65,28 @@ wea_c.entities = dofile(modpath.."/core/entities/init.lua") -- AFTER pos
 dofile(modpath.."/core/pos_marker_manage.lua") -- AFTER pos, entities
 dofile(modpath.."/core/pos_marker_wall_manage.lua") -- AFTER pos, entities
 
+-- Cleanup any existing nil entries in registered commands tables to prevent crashes
+local function cleanup_nil_entries()
+	-- Clean up WorldEditAdditions registered commands
+	for cmdname, def in pairs(wea_c.registered_commands) do
+		if def == nil then
+			wea_c.registered_commands[cmdname] = nil
+		end
+	end
+	
+	-- Clean up WorldEdit registered commands if available
+	if minetest.global_exists("worldedit") and worldedit.registered_commands then
+		for cmdname, def in pairs(worldedit.registered_commands) do
+			if def == nil then
+				worldedit.registered_commands[cmdname] = nil
+			end
+		end
+	end
+end
+
+-- Run cleanup immediately
+cleanup_nil_entries()
```
this check will run before any commands have been registered, since `worldeditadditions` (which actually registers commands) is loaded *after* `worldeditadditions_core`

sbrl reviewed (commented):
oops, we didn't realise that the `name` field on cmds was required! thanks so much for submitting this pull request to correct the issue.

I do note that some extra checks are present which don't do anything because of the way WEA is designed. Could you review my comments below at all please?

Many thanks for submitting this PR!

Commit:
Remove unnecessary cleanup code based on PR review

- Remove premature cleanup function that runs before commands are registered
- Revert unnecessary alias cleanup that occurs in error paths before registration
- Keep core fix: ensure name field is set in command definitions

sbrl reviewed (approved):
Many thanks!

sbrl commented:
Looks like you're trying ot merge into main rather than `dev`, so I'm going to have to merge this manually

sbrl commented:
Merged as ac24623, so while I'm closing this PR it's because I have merged it into a different branch

sbrl commented:
no it's not github I merged it
